File: src/Backend/Inspections.Core/Domain/CheckListAggregate/CheckListParam.cs
using Inspections.Shared;
using System;

namespace Inspections.Core.Domain.CheckListAggregate
{
    public class CheckListParam:Entity<int>
    {
        public int? CheckListId { get; set; }
        public int? CheckListItemId { get; set; }
        public CheckListItem CheckListItem { get; set; }
        public CheckList CheckList { get; set; }
        public string Key { get; set; }
        public string Value { get; set; }
        public CheckListParamType Type { get; set; }

        private CheckListParam() { } //Required by EF

        public CheckListParam(int? checkListId, int? checkListItemId, string key, string value, CheckListParamType type)
        {
            CheckListId = checkListId;
            CheckListItemId = checkListItemId;
            Key = key;
            Value = value;
            Type = type;
        }
    }
}

File: src/ClientApp/types/CheckLists/Models/CheckListParam.ts
import { CheckListItem } from './CheckListItem'
import { CheckList } from './CheckList'
import { CheckListParamType } from './CheckListParamType'

export interface CheckListParam {
    id: number;
    checkListId: number | null;
    checkListItemId: number | null;
    checkListItem: CheckListItem;
    checkList: CheckList;
    key: string;
    value: string;
    type: CheckListParamType;
}
